### ISSUE TITLE
Diagrams sprint

### ISSUE THREAD

**Issue by mrocklin** [dask/marketing#6]
We have a few diagrams that have been frequently used in docs, notebooks, slides, and so on.  These weren't particularly well made in the past, but they seem to have a great deal of value. 

@jrbourbeau and I may choose to sprint on making some more diagrams, in hopes that they improve our level of style, and also increase the impact that we have visually.

I think that there are two questions to answer before this sprint:

1.  What diagrams should we make?
2.  How should our style look like?

To start conversation, here are two existing drawings by @jrbourbeau , showing the client/scheduler/workers relationship and the collections/graphs/schedulers relationship:

![distributed-overview](https://user-images.githubusercontent.com/306380/66413985-27111600-e9be-11e9-9995-8f418ff48f8a.png)

![dask-overview (1)](https://user-images.githubusercontent.com/306380/66414592-5a07d980-e9bf-11e9-9e97-45dcffe418f7.png)

So stylistically we might ask questions like:

1.  What is our color palette?
2.  What fonts should we use
3.  Should we round corners, use bolder lines, ...

In terms of content we might consider the following diagrams (please feel free to edit)

- [x] Client/Scheduler/Worker relationship
- [x] collections -> graph -> schedulers
- [ ] What is a Dask worker (thread pool, data storage, connections to other workers, connections to scheduler)
- [x] Decide on a standardized dask array / dask dataframe image (there are a few lying around)
- [ ] How does Dask-XGBoost work?
- [x] scikit-learn & distributed joblib (the video https://ml.dask.org/joblib.html has an example diagram)
- [ ] Hyperband (prioritizes high-performing models, penalizes poor models. Check @stsievert's paper / scipy talk for an example).
- [x] The value of adaptivity
- [ ] Dask Gateway (cc @jcrist)
- [x] Deployment in general.  How does Dask relate to Kubernetes/Yarn/Job schedulers, and so on
- [ ] Beyond MapReduce.  Why dask delayed/futures are valuable
    [Slides from Mathew Lodge](https://www.slideshare.net/continuumio/cloud-native-data-science-mathew-lodge-strata-data-london?qid=29b54dc4-a3ee-44b0-845e-91adfd362f06&v=&b=&from_search=1) See 24-27.
- [ ] Dask futures and real-time processing
- [ ] Maybe something about the dashboard, and how it exposes scheduler state?
    [Here is a nice image that mixes a complex scheduler with steampunk](https://upload.wikimedia.org/wikipedia/commons/4/4a/Building_services_coordinated_drawing.JPG)
- [ ] Something for the Spark comparison page (high pageview count)
- [ ] Dask's relationship with GPUs
- [ ] Dependency diagram?  Maybe there is some way to highlight Xarray/RAPIDS/Prefect/...

I encourage others to add to this list.  If you have edit rights please exercise them.

cc @jrbourbeau @TomAugspurger @jcrist @birdsarah (in case you have time to briefly weigh in stylistically on the diagram above)

**Comment by jrbourbeau** [comment]
Thanks for getting the ball rolling on this @mrocklin! Here are a few thoughts I have regarding diagram styling. Would be interested to hear what others think too. 

> What is our color palette?

There are a few colors (ref https://marketing.dask.org/en/latest/colors.html) that match the documentation theme that we could also use in diagrams for overall consistency. A couple of thoughts that come to mind are (1) we should choose a non-orange-like color to help show visual distinction where necessary (e.g. collections and schedulers are different objects, so we could make them distinct colors too) and (2) the lighter orange color in those existing diagrams (#fda061) is a little intense and makes me want to squint my eyes

> What fonts should we use

FWIW, previously I've used [Inconsolata](https://fonts.google.com/specimen/Inconsolata). I like this font because I think it gives things an extra sense of styling and it reminds me of something you might see in terminal or text editor. That said, maybe we want these diagrams to be less distinctive and more familiar. In which case we could consider a more straightforward, cleaner font like [Roboto](https://fonts.google.com/specimen/Roboto) which is used widely on the web (Google [designed the font](https://github.com/google/roboto/) and uses it throughout their material design).

> Should we round corners, use bolder lines, ...

Personally, I'm a fan of straight corners, bolder lines, and a slight drop shadow below shapes (like the boxes in the above diagrams)

**Comment by mrocklin** [comment]
Also cc'ing @jacobtomlinson , who might find this kind of work of interest.

**Comment by jacobtomlinson** [comment]
As with all design related things consistency is key and having a set of rules to follow is really helpful because I'm generally not very good at remembering this stuff.

> What is our color palette?

I am definitely a fan of having a defined secondary color palette. Having the multiple shades of orange is nice as it allows you to break things up a little, but using them to draw clear distinctions may not be great for accessibility as they are very similar. Perhaps having official shades of green, purple or brown would be helpful. 

Looking through some corporate brand guidelines there seem to be a few takes on this. [Twitter](https://about.twitter.com/content/dam/about-twitter/company/brand-resources/en_us/Twitter_Brand_Guidelines_V2_0.pdf) have a selection of blue tinted greys to go with their corporate blue. [Netflix](https://brand.netflix.com/en/assets/brand-symbol/) seem to have no preference provided the brand red has a 3:1 contrast ratio with the color. [Spotify](https://developer.spotify.com/branding-guidelines/) only allow green, black and white, as does [Medium](https://www.behance.net/gallery/7226653/Medium-Brand-Development). [Uber](https://brand.uber.com/guide#color-secondary-colors) seem to have the widest range of secondary colors with a green, yellow, red, orange, brown and purple.

> What fonts should we use

Like both Inconsolata and Roboto. I think the diagrams above look great, especially the second one so my preference would propbably be Inconsolata.

I notice the docs use [Lato](https://fonts.google.com/specimen/Lato) and the blog uses Helvetica/Arial.

My preference would be to use a different font in diagrams to the body text when they are displayed in documentation and articles. But we should be careful to keep the number of fonts we use to a minimum. Consistency is good.

> Should we round corners, use bolder lines, ...

I share @jrbourbeau's preferences for straight corners, bold lines and subtle drop shadows. 

---

I also think we should make an effort to create SVG versions of diagrams as well as PNG for backward compatibility. Being able to rescale or edit an SVG makes them easier to reuse in other mediums (presentations, on websites, etc).

**Comment by jacobtomlinson** [comment]
Perhaps the blue and pink from this diagram are the kind of thing we could include on this palette.

![](https://docs.dask.org/en/latest/_images/dask-overview.svg)

There is also this blue that pops up in other parts of the documentation.

![](https://docs.dask.org/en/latest/_images/dask-array-black-text.svg)

**Comment by mrocklin** [comment]
I think it's good to be inspired by previous color choices, but I also think that we shouldn't be constrained by them.  They were chosen somewhat randomly while I was futzing about in inkscape. 

We have relatively few images out there.  It should be easy for us to replace all of them with a new color scheme.  I would start from the Dask Orange, and then find a good color to complement that.

**Comment by mrocklin** [comment]
Here is a diagram describing what makes up a Dask Worker: https://docs.google.com/drawings/d/1yAOdTiMrJb3xOSaxZsvW2UOXbWtbgr3EoolEPQ1DxMs/edit?usp=sharing

**Comment by mrocklin** [comment]
![Dask Worker](https://user-images.githubusercontent.com/306380/66517550-f1952700-eaa8-11e9-868c-41f6a98e101e.png)

Also, I have to say that Google Drawings is nice, but not as nice as Inkscape.  It's probably still a better choice though, just because we can all easily access it.

**Comment by mrocklin** [comment]
The inconsolata font, or any mono-spaced font, feels either intentionally retro, or unintentionally outdated.  Personally I would prefer that we use a non-mono-spaced font.

**Comment by mrocklin** [comment]
Especially because, I think, that these images will have proportionally more impact on those that are not programmers-first.

**Comment by mrocklin** [comment]
Here is another: https://docs.google.com/drawings/d/1AoEHphAfhgokORFsh_aZ33bD52Vqx5G2JWcgElk0nzY/edit?usp=sharing

![user-to-hardware](https://user-images.githubusercontent.com/306380/66522608-74bb7a80-eab3-11e9-8aca-e00cb3a98a7e.png)

**Comment by jacobtomlinson** [comment]
I've had a go at putting together some branding guidelines based on the discussion here. It just gives a framework for colors and fonts to work within. The idea is that this should reduce decision fatigue when creating content as you can just draw from this and also ensure some level of consistency.

I've done my best to check color combinations with a [contrast checker](https://webaim.org/resources/contrastchecker/) to ensure they are readable and accessible.

Would be keen to get feedback and iterate on this.

![Dask Branding Guidelines](https://user-images.githubusercontent.com/1610850/66560183-be0cd800-eb4e-11e9-8efa-a20f8c3bb0ba.png)

https://docs.google.com/drawings/d/1EGJDwI82qvXZJwD6EKqnqWUsmVYKmQ5TAHSqDzimWu4/edit?usp=sharing

**Comment by mrocklin** [comment]
Nice!  Thanks @jacobtomlinson .  Two thoughts:

1.  That is a lot of secondary colors.  
2.  Ideally we have both an orange and a non-orange that work well on light and dark backgrounds.  If we look at the gradient in the logo we see that this is true for the oranges (the logo looks good regardless)

**Comment by jacobtomlinson** [comment]
> That is a lot of secondary colors.

Reasonable comment. The top three I took from existing drawings in this thread. The bottom three I generated more intensionally using a palette tool. Green and red have cultural meaning and are often used to denote good/bad, success/failure, etc. I also wanted a darker color which could be displayed as text on a white background (was thinking hyperlinks in documentation specifically with that blue).

Happy to prune back a bit. I think four is a common number of secondary colors to have in a palette. 

> Ideally we have both an orange and a non-orange that work well on light and dark backgrounds. If we look at the gradient in the logo we see that this is true for the oranges (the logo looks good regardless)

Yes ideally. However none of those oranges meet the [W3C accessibility guidelines](https://www.w3.org/TR/WCAG20/) for text on a white background (they do for graphical objects and user interface components, so are fine in the logo).

We would need something as dark as this for use as text. Which is darker than any element in the logo.
![image](https://user-images.githubusercontent.com/1610850/66573088-038ace80-eb6a-11e9-84f5-f8674abf4ba3.png)

**Comment by jcrist** [comment]
Not to derail the conversation, but I've also been thinking about trying a dark gray instead of the straight black our current website uses. Black backgrounds often feel too dark and can make things harder to read (I am not a designer, see https://ianstormtaylor.com/design-tip-never-use-black/, https://uxmovement.com/content/why-you-should-never-use-pure-black-for-text-or-backgrounds/, https://graphicdesign.stackexchange.com/questions/25356/why-not-use-pure-black-000-and-pure-white-fff, etc...).

Instead of black, I'd suggest a dark gray e.g. `#1d1d1d`. Here's our website with this small change:

![Screenshot_2019-10-10 Dask — Dask 2 5 2 documentation](https://user-images.githubusercontent.com/2783717/66575171-1afe9300-eb3b-11e9-950a-fe7b1b054d7f.png)

**Comment by jacobtomlinson** [comment]
Yeah that's a good point @jcrist. 

From one of the things you linked:

> You usually want a high contrast between text and its background color. But too high contrast between design elements might give an unsettled and messy impression. Black and white create the highest contrast possible. 

The only counterpoint to this is with OLED displays where true black literally means "no light" from the LEDs and therefore `#000000` looks much better on those screens, especially in low light situations. One solution to this is to darken your white instead which reduces the contrast between the two. 

However ignoring that my preference is for a lighter black rather than a darker white.

**Comment by jacobtomlinson** [comment]
Here is an update with `#1D1D1D`

![Dask Branding Guidelines(1)](https://user-images.githubusercontent.com/1610850/66576303-692d8980-eb6f-11e9-83ee-f3c42e33165c.png)

**Comment by jacobtomlinson** [comment]
Here's another version with the lighter blacks.

![Dask Branding Guidelines(2)](https://user-images.githubusercontent.com/1610850/66576688-11435280-eb70-11e9-95b1-ef96d36d1547.png)

**Comment by jacobtomlinson** [comment]
_Sorry @mrocklin and @jrbourbeau we are way off track with the Diagrams sprint here, should we move this to another issue?_

**Comment by mrocklin** [comment]
I'm happy either way

**Comment by mrocklin** [comment]
I'm mostly just excited that we have a bunch of computational science
people talking about colors :)

**Comment by jcrist** [comment]
I personally prefer the `#1D1D1D` background. I'm happy to update the sphinx theme accordingly if I get the go-ahead. Also happy to open a new issue for this.

**Comment by mrocklin** [comment]
@jrbourbeau want to set some time aside tomorrow to make some diagrams together?  Establishing style ahead of time is ideal, but we can also modify things easily after the fact.

**Comment by TomAugspurger** [comment]
Here's a joblib / dask prototype

https://docs.google.com/drawings/d/1tWbey77qTozMrBenAOcjTFjpm2VIj0Gy0unHk5b5idc/edit?usp=sharing

![Screen Shot 2019-10-16 at 12 02 06 PM](https://user-images.githubusercontent.com/1312546/66941577-ceb8c480-f00c-11e9-99e1-49bfc4d981a9.png)

**Comment by jrbourbeau** [comment]
As discussed in the maintainer call yesterday, we're going to meet tomorrow (Oct 17) to make some diagrams (woo!). Let's meet in https://whereby.com/dask-dev from 8 AM - 12 PM central time (14:00 - 18:00 in the UK I think). Anyone who would like to participate should feel free to join for whatever portion of time you're available.

**Comment by TomAugspurger** [comment]
Did the topic of different backgrounds come up yesterday? Say I have a diagram I'd like to use on both docs.dask.org (light background) and in the Dask slide deck (dark background). Is there an easy way to do that without too much duplication?

**Comment by jacobtomlinson** [comment]
Yeah it came up. I think the plan was to provide both, at least for highly used diagrams.

**Comment by jsignell** [comment]
I just used the colors provided in https://github.com/dask/marketing/issues/6#issuecomment-540605614 in https://github.com/dask/dask/issues/5965. Should that branding image be added to https://marketing.dask.org. Or is it already and I just didn't find it?

**Comment by jrbourbeau** [comment]
Moving to marketing.dask.org sounds good to me. Currently I think the style guide hasn't moved outside of the issue you linked to @jsignell. Perhaps the "Style and Colors" page is a natural place for it
